compiler-plugin/src/main/java/arrow/plugin/MetaClassBuilderInterceptorExtension.kt

```diff
@@ -55,11 +55,11 @@ internal class MetaClassBuilder(
       !functionName.startsWith("<set")) {
       val functionPsi = function.findPsi()
       //if the function returns Unit then it should have been suspended since all it can do is produce effects
-      if (function.returnType?.isUnit() == true) {
+      if (function.returnType?.isUnit() == true || function.returnType?.isNothingOrNullableNothing() == true) {
```

Comment:
You might also clarify that methods returning Nothing are also considered to contain side effects in the comment above.